Don't call g_thread_unref after g_thread_join, g_thread_join already calls g_thread_unref.
https://docs.gtk.org/glib/struct.Thread.html

From 3990212ac5e22ed8f9f6f83c76de0ec3651564c9 Mon Sep 17 00:00:00 2001
From: Jens Georg <mail@jensge.org>
Date: Sat, 29 Mar 2025 12:29:48 +0100
Subject: [PATCH] tests: Remove unnecessary g_thread_unref

---
 tests/test-service-proxy.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/tests/test-service-proxy.c b/tests/test-service-proxy.c
index cc3c613..988e0e6 100644
--- a/tests/test-service-proxy.c
+++ b/tests/test-service-proxy.c
@@ -517,8 +517,6 @@ test_sync_call (ProxyTestFixture *tf, gconstpointer user_data)
         // Spin the loop for a bit...
         g_timeout_add (500, (GSourceFunc) delayed_loop_quitter, tf->loop);
         g_main_loop_run (tf->loop);
-
-        g_thread_unref (t);
 }
 
 gboolean
@@ -562,7 +560,6 @@ test_cancel_sync_call (ProxyTestFixture *tf, gconstpointer user_data)
         // Spin the loop for a bit...
         g_timeout_add (500, (GSourceFunc) delayed_loop_quitter, tf->loop);
         g_main_loop_run (tf->loop);
-        g_thread_unref (t);
         g_object_unref (d.cancellable);
 }
 
@@ -642,7 +639,6 @@ test_finish_soap_error_sync (ProxyTestFixture *tf, gconstpointer user_data)
         // Spin the loop for a bit...
         g_timeout_add (500, (GSourceFunc) delayed_loop_quitter, tf->loop);
         g_main_loop_run (tf->loop);
-        g_thread_unref (t);
 }
 
 void
-- 
GitLab

